openfisca_core/formulas.py

```diff
@@ -663,6 +620,50 @@ def get_neutralized_column(column):
         )
 
 
+def extract_formula_date_from_name(formula_name, formula_name_prefix, formula_name_separator):
+    """
+    Return a day date extracted from formula name.
+    Valid dated name formats are : 'formula', 'formula_YYYY', 'formula_YYYY_MM' and 'formula_YYYY_MM_DD' where YYYY, MM and DD are a year, month and day.
+
+    Default year is '0001'. Default month and day are '01'.
+    """
+    formula_default_year = '0001'
+    formula_default_month = '01'
+    formula_default_day = '01'
+
+    if formula_name == formula_name_prefix:
+        formula_name += formula_name_separator + formula_default_year + formula_name_separator\
+            + formula_default_month + formula_name_separator\
+            + formula_default_day
+
+    else:
+        match = re.match(r'(formula_)(\d{4}(_\d{2}){0,2})', formula_name)  # YYYY or YYYY_MM or YYYY_MM_DD
+        assert match, 'Unrecognized formula name. Expecting "formula_YYYY" or "formula_YYYY_MM" or "formula_YYYY_MM_DD where YYYY, MM and DD are year, month and day. Found: ' + formula_name
+        start_str = match.group(2)
+
+        if len(start_str) == 4:  # YYYY
+            start_str += formula_name_separator + formula_default_month
+            formula_name += formula_name_separator + formula_default_month
+
+        if len(start_str) == 7:  # YYYY_MM
+            start_str += formula_name_separator + formula_default_day
+            formula_name += formula_name_separator + formula_default_day
+
+    return datetime.datetime.strptime(formula_name, formula_name_prefix + formula_name_separator + '%Y_%m_%d').date()
+
+
+def get_datetime_date(variable_name, date_str):
+    """
+    Return a datetime.date from a date string.
+    """
+    try:
+        time = datetime.datetime.strptime(date_str, '%Y-%m-%d')
+    except ValueError:
+        raise ValueError(u"Incorrect 'end' attribute format in '{}'. 'YYYY-MM-DD' expected where YYYY, MM and DD are year, month and day. Found: {}".format(variable_name, date_str).encode('utf-8'))
```

Comment:
Why does a function named `get_datetime_date` know about an `end` attribute of a variable ? This error should be thrown directly by `new_filled_column`.